lib/Sema/SemaExpr.cpp

```diff
@@ -394,11 +394,18 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, SourceLocation Loc,
   // type in checked block.
   // Otherwise, it does not emit redundant error message at the use of variable
   // since it already produced an error message at the declaration of variable.
+  // Calls to variable argument functions are not allowed in a checked block.
```

Comment:
You can declare a function pointer that takes variable arguments and has a variable argument type.   I think you need check this at the type level too, just as you a check for uses of unchecked pointer types.

This makes me wonder about what we should do when you have constructed types in a checked block that use unchecked pointer types.  For example, you could have a pointer to an unchecked pointer.    Checked code could never use the unchecked pointer, but it could pass it around.